examples/kitchen-sink.yaml

```diff
@@ -434,3 +434,107 @@ resource:
         - process.command_args
   # Configure the resource schema URL.
   schema_url: https://opentelemetry.io/schemas/1.16.0
+
+# Configure instrumentation.
+instrumentation:
+  # Configure general SemConv options that may apply to multiple languages and instrumentations.
+  #
+  # Instrumenation may merge general config options with the language specific configuration at .instrumentation.<language>.
+  general:
+    # Configure instrumentations following the peer semantic conventions.
+    # 
+    # See peer semantic conventions: https://opentelemetry.io/docs/specs/semconv/attributes-registry/peer/
+    peer:
+      # Configure the service mapping for instrumentations following peer.service semantic conventions.
+      #
+      # Each entry is a key value pair where "peer" defines the IP address and "service" defines the corresponding logical name of the service.
+      # 
+      # See peer.service semantic conventions: https://opentelemetry.io/docs/specs/semconv/general/attributes/#general-remote-service-attributes
+      service_mapping:
+        - peer: 1.2.3.4
+          service: FooService
+        - peer: 2.3.4.5
+          service: BarService
+    # Configure instrumentations following the http semantic conventions.
+    # 
+    # See http semantic conventions: https://opentelemetry.io/docs/specs/semconv/http/
+    http:
+      # Configure instrumentations following the http client semantic conventions.
+      client:
+        # Configure headers to capture for outbound http requests.
+        request_captured_headers:
```

Comment:
😢 I like hyphens, but that's probably just a Java thing...

Reply:
The casing of property keys should be consistent. While the decision of what case is somewhat arbitrary and subject to bikeshedding, but I think lower_snake_case is the best choice because its consistent with how the collector models YAML configuration today.

Opened #96 to codify this in naming rules.